src/transactions/EventManager.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+// Copyright 2025 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "main/Config.h"
+#include "xdr/Stellar-ledger-entries.h"
+#include "xdr/Stellar-ledger.h"
+#include "xdr/Stellar-transaction.h"
+
+namespace stellar
+{
+class EventManager;
+using EventManagerPtr = std::shared_ptr<EventManager>;
+class TransactionFrameBase;
+using TransactionFrameBaseConstPtr =
```

Comment:
This `using` re-declaration seems sketchy. Can we just store const reference to the parent tx? I don't think event manager should outlive the tx frame.

Reply:
done